### PR TITLE
Improve TOC browsing

### LINKED ISSUE #285: Restructure table of contents
To make the chapters and recipes easier to browse I think it would be useful to list them all on the front-page README. Right now it is a bit hard to know which chapter contains what without a lot of clicking.

We can do this after we have reshuffled the chapters.

### PR DESCRIPTION
**Pull request by bast**
Motivation:
- easier browsing
- better consistency
- less manual code duplication
- easier refactoring

Fixes #285.

### CODE REVIEW

**Comment by robertodr** [comment]
Can you summarize a bit how this works and what should be done to modify the READMEs? The diff is overwhelming :sweat_smile:

**Comment by bast** [comment]
The diff is huge because I renamed paths but you need to run:
https://github.com/bast/cmake-recipes/blob/radovan/improve-toc-browsing/contributing/README.md#updating-the-table-of-contents-and-readme-files

And here it's how it will look:
https://github.com/bast/cmake-recipes/blob/radovan/improve-toc-browsing/README.md

**Comment by bast** [comment]
Perhaps I should add to the README that the READMEs are created from title.txt and abstract.md files.

**Comment by robertodr** [comment]
OK, thanks for the explanation. And yes, add a note on what to touch when updating.
